tests/test_system.py

```diff
@@ -46,21 +62,41 @@ def test_schema_endpoint_login_details(client):
     login_schema = body["login"]
     presigned_url_schema = body["presigned_url"]
 
-    assert login_schema["version"] == 2
+    assert login_schema["version"] == 2.0
     model = login_schema["model"]
 
+    optional_login_fields = ["fence_idp", "shib_idp", "client_id", "ip"]
     for field in log_fields + login_fields:
         assert field in model
+        if field in optional_login_fields:
+            assert model[field].endswith("?")
 
-    assert model.get("ip", "").endswith("?")
-
-    assert presigned_url_schema["version"] == 1
+    assert presigned_url_schema["version"] == 1.0
     model = presigned_url_schema["model"]
 
     for field in log_fields + presigned_url_fields:
         assert field in model
 
 
+def test_category_version_schema_alignment(client):
+    """
+    Tests that every category declared in CATEGORY_TO_MODEL_CLASS:
+    - Has corresponding entry in CURRENT_SCHEMA_VERSIONS
+    - Appears in the /_schema endpoint response with the same version.
+    """
+    assert set(CATEGORY_TO_MODEL_CLASS) == set(CURRENT_SCHEMA_VERSIONS)
+
+    resp = client.get("/_schema")
+    assert resp.status_code == 200, resp.text
+    schema_body = resp.json()
+
+    assert set(schema_body) == set(CATEGORY_TO_MODEL_CLASS)
```

Comment:
could you add an error message to the 2 `CATEGORY_TO_MODEL_CLASS` assertions, something like "Expected all categories in X to have an entry in Y"